src/include/storage/write_ahead_log/log_manager.h

```diff
@@ -28,15 +32,31 @@ class LogManager {
    *                    buffers from
    */
   LogManager(const char *log_file_path, RecordBufferSegmentPool *const buffer_pool)
-      : out_(log_file_path), buffer_pool_(buffer_pool) {}
+      : buffer_pool_(buffer_pool),
+        buffer_to_write_(nullptr),
+        empty_buffer_queue_(MAX_BUF),
+        filled_buffer_queue_(MAX_BUF),
+        run_log_writer_thread_(true),
+        do_persist_(true) {
+    for (int i = 0; i < MAX_BUF; i++) {
+      buffers_.emplace_back(BufferedLogWriter(log_file_path));
+    }
+    for (int i = 0; i < MAX_BUF; i++) {
+      BlockingEnqueueBuffer(&buffers_[i], &empty_buffer_queue_);
+    }
+    log_writer_thread_ = std::thread([this] { WriteToDiskLoop(); });
```

Comment:
We don't want to couple log thread start with log manager object creation. Eventually we want to link every thread to a central registry for self-driving stuff. Can you provide start methods instead? (See GC for example)